ha/fault_tolerance/rest_api.py

```diff
@@ -0,0 +1,255 @@
+
+    @staticmethod
+    async def _on_startup(app):
+        """
+        This function is executes on startup of CC Rest API Server.
+        Args:
+            app (web.Application): web Application object which is starting.
+        """
+        Log.info(f'REST API server {CcRestApi._site.name} startup')
+        # Note: Additional calls needs to be added to execute on startup of CC Rest API Server
+
+    @staticmethod
+    async def _on_shutdown(app):
+        """
+        This function is executes on shutdown of CC Rest API Server.
+        Args:
+            app (web.Application): web Application object which is shuting down.
+        """
+        Log.info(f'REST API server {CcRestApi._site.name} shutdown')
+        # Note: Additional calls needs to be added to execute on shutdown of CC Rest API Server
+
+    @staticmethod
+    def _start_server(app, host: str=None, port: int=None, ssl_context: ssl.SSLContext=None, access_log=None):
+        """
+        Starts CC REST Application server.
+        Sets the signal_handler if CcRestApi._handle_signal is set to true in init().
+        Args:
+            app (web.Application): web Application object which is starting.
+            host (str, optional): CC REST API server host. Defaults to None.
+            port (int, optional): CC REST API server port. Defaults to None.
+            ssl_context (ssl.SSLContext, optional): SSL context object for 'https' protocol. Defaults to None.
+            access_log (any, optional): access log. Defaults to None.
+        """
+        CcRestApi._loop = asyncio.get_event_loop()
+        runner = web.AppRunner(app, handle_signals=CcRestApi._handle_signals, access_log=access_log)
+        CcRestApi._loop.run_until_complete(runner.setup())
+        CcRestApi._site = web.TCPSite(runner, host=host, port=port, ssl_context=ssl_context)
+        CcRestApi._loop.run_until_complete(CcRestApi._site.start())
+        Log.info(f'======== CC REST API Server is running on {CcRestApi._site.name} ========')
+
+        # Add signal handlers
+        if CcRestApi._handle_signals:
+            for signame in  CcRestApi._signals:
+                CcRestApi._loop.add_signal_handler(getattr(signal, signame),
+                                    lambda signame=signame: asyncio.ensure_future(CcRestApi.handle_signal(signame)))
+
+    @staticmethod
+    def stop():
+        """
+        Stops REST API server.
+        Warpper to call async coroutine CcRestApi._stop().
+        """
+        Log.info(f'CC Rest API Server {CcRestApi._site.name} is stopping.')
+        CcRestApi._loop().run_until_complete(CcRestApi._stop())
+
+    @staticmethod
+    def start():
+        """
+        Starts CC REST API server.
+        fetches the configuration from conf and starts server.
+        """
+        host = None
+        ha_endpoint = Conf.get(const.HA_GLOBAL_INDEX, f'service_config{const.HA_DELIM}endpoint')
+        if ha_endpoint:
```

Comment:
What if there will be some problem in fetching port from config?
I think this should be handled in try catch.

Reply:
Yes, As it's an enhancement will take care this handling in my task.

Reply:
Ok. Thanks!